pkg/kallsyms/kallsyms.go

```diff
@@ -137,6 +137,8 @@ var (
 	triedGetAddr = map[string]error{}
 )
 
+type symbolResolver func(symbol string) (uint64, error)
```

Comment:
I'm wondering if defining it as an interface and implementing it on a Kallsyms and Bpf resolvers will make the code cleaner.